recipes/pcapplusplus/22.11/conanfile.py

```diff
@@ -0,0 +1,124 @@
+from conan import ConanFile
+from conan.tools import files
+from conan.tools.microsoft import is_msvc
+from conan.errors import ConanInvalidConfiguration
+from conans import AutoToolsBuildEnvironment, tools
+# It has to be this one, the one from conan.tools.microsoft does not pass toolset
+from conans import MSBuild
+import os
+
+required_conan_version = ">=1.50.0"
+
+
+class PcapplusplusConan(ConanFile):
+    name = "pcapplusplus"
+    license = "Unlicense"
+    description = "PcapPlusPlus is a multiplatform C++ library for capturing, parsing and crafting of network packets"
+    topics = ("conan", "pcapplusplus", "pcap", "network", "security", "packet")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/seladb/PcapPlusPlus"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False],
+        "immediate_mode": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+        "immediate_mode": False,
+    }
+
+    exports_sources = "patches/*"
+    generators = "make"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.settings.os == "Windows":
+            self.requires("npcap/1.70")
+        else:
+            self.requires("libpcap/1.9.1")
+
+        # TODO: use conan recipe instead of embedded one
+        # self.requires("hash-library/8.0")
+
+    def validate(self):
+        if self.settings.os == "Windows" and not is_msvc(self):
+            raise ConanInvalidConfiguration("Can not build on Windows: only msvc compiler is supported.")
+        if self.settings.os not in ("FreeBSD", "Linux", "Macos", "Windows"):
+            raise ConanInvalidConfiguration(f"{self.settings.os} is not supported")
+
+    def source(self):
+        files.get(self, **self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    @property
+    def _configure_sh_script(self):
```

Comment:
This goes into the `generate()` method in conan 2.0. Also, I think it would be my preference to explicitly enumerate the configure args rather than deferring to a script because it makes it clearer what the recipe is actually doing, but I'll let a maintainer chime in on whether this is acceptable practice.